.github/workflows/on_push.yaml

```diff
@@ -2,8 +2,6 @@ name: CI workflow on push
 
 on:
   push:
-    branches:
```

Comment:
Do we want to run on every push to any branch even if there's no PR created?

Reply:
IMO yes, because ideally, we shouldn't be creating PRs with failing tests.

Reply:
I don't understand why we would. On PR the CI will run, if I push changes to a random branch which doesn't have a PR, just for the sake of saving local changes, the CI will run. Do we want this behavior? In the other projects we're following this approach https://github.com/canonical/sdcore-upf-operator/blob/main/.github/workflows/main.yaml

Let me know if that makes sense, if doesn't feel free to resolve, I don't have a strong opinion honestly.

Reply:
My point is that in the previous configuration, the first moment you'll find out that your tests are failing (and we all know local runs can be misleading) will be **after** you create a PR. This means that you'll, most likely, need to change something in the code, so technically, until that's done, your change isn't ready for review. As I mentioned in the previous comment, IMO, ideally we should be creating PRs when everything is squared away. 
As for saving local changes, you can simply commit without pushing or push and ignore the fact that the tests will fail.